application/client/src/app/ui/views/sidebar/search/filters/provider.ts

```diff
@@ -297,38 +319,37 @@ export class ProviderFilters extends Provider<FilterRequest> {
                 }
                 const disabled: DisabledRequest = outside.extract();
                 this.session.search.store().disabled().delete([disabled.uuid()]);
-                this.session.search.store().filters().tryRestore(disabled.entity());
-                // if (data.entries !== undefined) {
-                //     const outside: Entity<ChartRequest> | undefined =
-                //         data.entries[event.previousIndex] !== undefined
-                //             ? (data.entries[index] as Entity<ChartRequest>)
-                //             : undefined;
-                //     if (
-                //         outside !== undefined &&
-                //         typeof outside.getEntity === 'function' &&
-                //         outside.extract() instanceof ChartRequest
-                //     ) {
-                //         session
-                //             .getSessionSearch()
-                //             .getChartsAPI()
-                //             .getStorage()
-                //             .remove(outside.extract());
-                //         session
-                //             .getSessionSearch()
-                //             .getFiltersAPI()
-                //             .getStorage()
-                //             .add(
-                //                 {
-                //                     request: outside.extract().asDesc().request,
-                //                     flags: {
-                //                         casesensitive: true,
-                //                         wholeword: true,
-                //                         regexp: true,
-                //                     },
-                //                 },
-                //                 event.currentIndex,
-                //             );
-                //     }
+                this.session.search
+                    .store()
+                    .filters()
+                    .update([disabled.entity() as StoredEntity<FilterRequest>]);
+            } else {
+                if (data.entries !== undefined && data.entries.length > 0) {
+                    const entry: Entity<ChartRequest> = data.entries[0] as Entity<ChartRequest>;
+                    const chartRequest: ChartRequest = entry.extract();
+                    this.session.search.store().filters().delete([chartRequest.uuid()]);
+                    this.session.search
+                        .store()
+                        .filters()
+                        .update([
+                            new FilterRequest({
```

Comment:
Update method gets `StoredEntity<T>`, but you are sending `FilterRequest`. That's mistake